src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -937,7 +937,7 @@ InferenceEngine::Blob::Ptr InferRequest::create_device_blob(const InferenceEngin
 
     auto l = cldnn::layout(shape, dt, format);
 
-    if (m_graph->GetEngine()->use_unified_shared_memory()) {
+    if (m_graph->GetEngine()->use_unified_shared_memory() && !m_graph->GetNetwork()->is_dynamic()) {
```

Comment:
1) If dynamic, will it go to the else path?
=> so, it will allocate remoteCLbuffer? is this your expectation? 

2) As I fixed in the PR https://github.com/openvinotoolkit/openvino/pull/13306, there is the case where the graph is dynamic, but the network input or output might be static, in such a case, the memory should be allocated.

Reply:
The problem is already solved by your PR https://github.com/openvinotoolkit/openvino/pull/13306